File: models/Movie.js
// models/Movie.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const MovieSchema = new Schema({
  title:        { type: String, required: true },
  description:  String,
  duration:     Number,
  director:     String,
  cast:         [String],
  genre:        [String],
  release_date: Date,
  language:     String,
  poster:       String,
  rating:       Number,

  // ⭐ GIỚI HẠN ĐỘ TUỔI
  // 0 = phổ biến, 13 = T13, 16 = T16, 18 = T18
  age_limit:    { type: Number, default: 0 },

  status:       { type: String, enum: ['coming','now_showing','archived'], default: 'coming' },
  created_at:   { type: Date, default: Date.now }
});

module.exports = mongoose.model('Movie', MovieSchema);

File: app/src/main/java/com/example/datn_md_13/Model/Movie.java
package com.example.datn_md_13.Model;

import com.google.gson.annotations.SerializedName;
import java.util.Date;
import java.util.List;

public class Movie {
    @SerializedName("_id")
    public String id;

    public String title;
    public String description;
    public Integer duration;       // minutes
    public String director;
    public List<String> cast;
    public List<String> genre;

    @SerializedName("release_date")
    public Date releaseDate;       // ngày khởi chiếu

    public String language;
    public String poster;          // URL ảnh
    public Double rating;

    // coming | now_showing | archived
    public String status;

    @SerializedName("created_at")
    public Date createdAt;
}
